### DISCUSSION TITLE
How to suggest update for invalid version ?

### DISCUSSION THREAD

**Discussion by azro352** [renovatebot/renovate#25062] [Request Help] [answered]
> Self-hosted v37.2.0-full // GitLab self-hosted

Hi, I've bad named docker in projects, which uses rolling tag, like the horrible `nginx:stable`

Is it possible to hardcode in the config that if I see `nginx:stable` I want to suggest the version `1.25.1-alpine-slim` (don't care about the latest, renovate will suggest it later) ?

That will allow me to activate renovate on dozen of project and easily upgrade them from that bag naming 

Thanks

**Comment by rarkins** [comment]
You can maybe do it with the replacement functionality. Examples in https://docs.renovatebot.com/presets-replacements/

**Reply by azro352** [reply]
Just tried that, the docker is skipped because of "invalid-value" (currentValue being "stable")

Do I need to "authorize" this strange version somewhere ? Doesn't replacement execute before the main workflow ?

**Reply by rarkins** [reply]
I'm not sure, you might need to do something hacky with the regex manager so that you can rewrite/override the value

**Comment by azro352** [comment, marked as answer]
I failed to fix it with `matchCurrentVersion` but succeeded with `matchCurrentValue`

```json
{
  "packageRules": [
    {
      "matchDatasources": ["docker"],
      "matchPackageNames": ["custom.apifactory.url/docker/nginx"],
      "matchCurrentValue": "/^stable$/",
      "replacementName": "other.apifactory.url/docker/nginx",
      "replacementVersion": "1.25.2-alpine-slim"
    },
  ]
}
```
